src/pepr/operator/controllers/keycloak/authservice/config.ts

```diff
@@ -99,47 +130,112 @@ function buildInitialSecret(): AuthserviceConfig {
   return config;
 }
 
-export async function getAuthserviceConfig() {
-  const authSvcSecret = await K8s(kind.Secret)
+/**
+ * Retrieves the authservice configuration, either from the in-memory cache
+ * or from the Kubernetes secret if not already cached.
+ *
+ * @returns {Promise<AuthserviceConfig>} - The authservice configuration.
+ */
+export async function getAuthserviceConfig(): Promise<AuthserviceConfig> {
+  if (inMemorySecret) {
+    log.info("Returning in-memory authservice secret");
+    return Promise.resolve(inMemorySecret);
+  }
+
+  // Fetch the authservice secret from Kubernetes if not in cache
+  pendingSecretFetch = K8s(kind.Secret)
```

Comment:
In the event of multiple simultaneous calls to this function, i think we could end up with multiple calls to fetch the secret if there isn't one in the cache already. Checking before could be beneficial **i think**. 
```suggestion
if (pendingSecretFetch) {
  return pendingSecretFetch;
}
pendingSecretFetch = K8s(kind.Secret)
...
```